### PR TITLE
DataNode: Setting the remote reindexing allowlist separately

### PR DESCRIPTION
**Pull request by janheise**
## Description

The allowlist needs to include all machines that we re-index from. Also, the setting can contain wildcards so we need a separate setting for the host we execute the action against.

/nocl

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by bernd — COMMENTED** [review]
It would be great to avoid the non-inclusive term and use the `reindex.remote.allowlist` setting instead. (and rename our methods)

**Comment by bernd** [comment]
Closes https://github.com/Graylog2/graylog2-server/issues/18758

File: graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java [hunk 1]
```diff
@@ -196,12 +196,13 @@ public boolean dataNodeStartupFinished() {
 
     @Override
     public void startRemoteReindex() {
+        final String allowlist = getStateMachineContext().getActionArgumentOpt("allowlist", String.class).orElseThrow(() -> new NullPointerException("allowlist has tp be provided"));
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java:199]: The form will return an empty string if no setting is provided, so the validation does not really work. Also, there's a typo (has t`o`)

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx [hunk 2]
```diff
@@ -108,15 +109,25 @@ const MigrateExistingData = ({ currentStep, onTriggerStep }: MigrationStepCompon
...
-    <Formik initialValues={initialValues} onSubmit={() => {}}>
+    <Formik initialValues={initialValues}
+            onSubmit={() => {
+            }}>
       {({
         values,
         setFieldValue,
       }) => (
         <Form role="form">
+          <Input id="allowlist"
```
Comment (moesterheld) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx:120]: we need to provide more info in this form to make it easier to understand, e.g. example values

**Review by moesterheld — APPROVED** [review]
Successfully tested migration of a single node instance.
Three comments, but I would address those in a separate pr to not hold back merging this one

File: graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java (hunk 1 above)
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java:199]: but I tagged the field as "required" in the form?

Comment (moesterheld) [reply graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java:199]: hmm.. i could submit it with an empty value

**Pull request merged**